src/summary_writer.jl

```diff
@@ -76,7 +99,30 @@ function Base.write(writer::FileWriter, graph::Graph)
     write(writer, event)
 end
 
+function set_default(writer::FileWriter)
+    context = tf.Context()
+    context.attrs["default_file_writer"] = writer
+    push!(tf.global_context, context)
+end
+
+function with_default(writer::FileWriter, block)
```

Comment:
```suggestion
function with_default(block, writer::FileWriter)
```

so that do-blocks work

Reply:
done. somehow always forget that julia takes the block first.